core/files/assets/files-upload-handler.php

```diff
@@ -89,7 +89,9 @@ public function handle_upload_prefilter( $file ) {
 	}
 
 	protected function is_file_should_handled( $file ) {
-		return $this->is_elementor_media_upload() && $this->get_mime_type() === $file['type'];
+		$ext = pathinfo( $file['name'], PATHINFO_EXTENSION );
+
+		return $this->is_elementor_media_upload() && $this->get_mime_type() === $file['type'] && $this->get_file_type() === $ext;
```

Comment:
Ask @rotemee if we can remove the mime-type check.

Reply:
Well, It doesn't hurt...